File: Java/euler.java
/*----------------------------------------------------------------
*
* Programación avanzada: Proyecto final
* Fecha: 30-Nov-2022
* Autor: A01704052 Martin Noboa
* Descripción: Implementacion secuencial de la serie de Euler en Java
*
*--------------------------------------------------------------*/
import java.util.*;

public class Euler {
	private static final int SIZE = 100;
	private double result,n,sign;

	public Euler() {
		this.n = 2;
		this.sign = 1;
		this.result = 3;
	}

	public double getResult() {
		return result;
	}

	public void calculate() {
		for (int i = 0; i <= 1000000; i++) {
			this.result = this.result + (this.sign * (4 / ((this.n) * (this.n + 1) * (this.n + 2))));
			this.sign = this.sign * (-1);
			this.n += 2;
		}
	}

	public static void main(String args[]) {
		long startTime, stopTime;
		double acum = 0;

		Euler e = new Euler();
		System.out.printf("Starting...\n");

		for (int i = 0; i < 10; i++) {
			startTime = System.currentTimeMillis();
			e.calculate();
			stopTime = System.currentTimeMillis();
			acum += (stopTime - startTime);
		}
		System.out.printf("Pi = %.8f\n", e.getResult());
		System.out.printf("avg time = %.10f ms\n", (acum / 10));
	}
}

File: C/euler.c
/*----------------------------------------------------------------
*
* Programación avanzada: Proyecto final
* Fecha: 30-Nov-2022
* Autor: A01704052 Martin Noboa
* Descripción: Implementacion secuencial de la serie de Euler en C
*
*--------------------------------------------------------------*/

#include <stdio.h>
#include <stdlib.h>
#include "utils.h"

#define SIZE 10000000

double euler(double n, double sign) {
  double pi = 0;
  for (int i = 0; i <= SIZE; i++) {
        pi = pi + (sign * (4 / ((n) * (n + 1) * (n + 2))));
        sign = sign * (-1);
        n += 2;
    }
 
    return pi + 3;
}

int main(int argc, char* argv[]) {
  double n = 2, sign =1;
  int i;
  double ms, result;
	printf("Starting...\n");
	ms = 0;
	for (i = 0; i < N; i++) {
		start_timer();
		result = euler(n, sign);
		ms += stop_timer();
	}
  printf("Pi =  %0.9lf\n",result);
	printf("avg time = %.5lf ms\n", (ms / N));
  
  return 0;
}
